test/test_sparse_csr.py

```diff
@@ -2627,29 +2627,42 @@ def test_compressed_layout_conversions_coverage(self, device, from_layout, to_la
         that an exception is thrown for unsupported conversions.
         """
 
-        def _to_from_layout(layout_a, layout_b):
-            a = make_tensor((6, 10), dtype=torch.float, device=device)
-            expect_error = (layout_a in [torch.sparse_csc, torch.sparse_bsc]
-                            or layout_b in [torch.sparse_csc, torch.sparse_bsc])
-            expect_error = expect_error or (layout_a, layout_b) == (torch.sparse_bsr, torch.sparse_bsr)
-            expect_error = expect_error or (layout_a, layout_b) == (torch.sparse_bsr, torch.sparse_csr)
-            # CSC to CSR conversion is supported
-            if layout_a is torch.sparse_csc and layout_b is torch.sparse_csr:
-                expect_error = False
-            # CSC to CSC conversion is supported
-            if layout_a is torch.sparse_csc and layout_b is torch.sparse_csc:
+        def _to_from_layout(layout_a, layout_b, a):
+            expect_error = True
+            allowed_layout_ab_sets = {
+                frozenset({torch.sparse_csc}),
+                frozenset({torch.sparse_csr}),
+                frozenset({torch.sparse_csc, torch.sparse_csr}),
+                frozenset({torch.sparse_bsc}),
+                frozenset({torch.sparse_bsr}),
+                frozenset({torch.sparse_bsc, torch.sparse_bsr}),
+                frozenset({torch.sparse_csr, torch.sparse_bsr}),
+            }
+            if {layout_a, layout_b} in allowed_layout_ab_sets:
                 expect_error = False
+
+            # BSR -> CSR is not yet supported
+            if (layout_a, layout_b) == (torch.sparse_bsr, torch.sparse_csr):
+                expect_error = True
+            # CSR -> BSR only works for non-batched inputs
+            if (layout_a, layout_b) == (torch.sparse_csr, torch.sparse_bsr):
+                if a.dim() > 2:
+                    expect_error = True
+
             if expect_error:
                 with self.assertRaises(RuntimeError):
                     b = self._convert_to_layout(a, layout_a)
```

Comment:
The test is targeting the `layout_a` -> `layout_b` conversion, so the `strided` -> `layout_a` conversion should probably move outside the context manager to ensure the target conversion is the one raising.

Reply:
yeah, that was there before, I did not introduce that part. But I totally agree, will change that.